src/backends/tensorrt/plan_backend.cc

```diff
@@ -2702,9 +2772,9 @@ PlanBackend::Context::Run(
     }
   }
 
-  // Ensure inputs are ready before execution. Output buffers will always be
-  // available at this point as the execution and output copy are on the same
-  // stream.
+  // Ensure inputs are ready before execution.
+  // Output buffers are guaranteed to be available at this point when the
+  // execution and output copy are on the same stream.
```

Comment:
How do we know that output buffers are available when we are using a separate output stream?

Reply:
Nice point! Updated to wait on the output buffer being available. This shouldn't cause any perf problems as we ping pong between buffers using the same inference stream.